### PR TITLE
Feat: describe origin command

### PR DESCRIPTION
**Pull request by maxwelbm**
- added describe origin command

### CODE REVIEW

File: pkg/cmd/describe/origin/origin.go [hunk 1]
```diff
@@ -28,24 +37,48 @@ var (
...
+				appID, err := strconv.Atoi(answers)
+				if err != nil {
+					logger.Debug("Error while parsing string to integer", zap.Error(err))
+					return utils.ErrorConvertingStringToInt
+				}
+
+				applicationID = int64(appID)
+			}
+
+			if !cmd.Flags().Changed("origin-id") {
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:63]: Origin needs origin-key to be deleted/described etc...

File: messages/describe/origin/errors.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package origin
+
+import "errors"
+
+var (
+	ErrorMissingArguments = errors.New("Required flags are missing. You must supply application-id and origin-id as arguments. Run 'azion <command> <subcommand> --help' command to display more information and try again")
```
Comment (PatrickMenoti) [inline messages/describe/origin/errors.go:6]: You can remove this. We never return this error, as we ask for inputs if not sent.

File: pkg/cmd/describe/origin/origin.go (hunk 1 above)
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:63]: I don't remember why we did it like this. However you can GET an origin by its origin_key.
https://api.azion.com/#850d2a1d-98c2-4c29-91a0-8173ddaac8da

File: pkg/cmd/describe/origin/origin.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ func NewCmd(f *cmdutil.Factory) *cobra.Command {
 				applicationID = int64(appID)
 			}
 
-			if !cmd.Flags().Changed("origin-id") {
+			if !cmd.Flags().Changed("origin-key") {
 				answers, err := utils.AskInput("What is the ID of the Origin?")
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:64]: Put this message in a variable. Also, you have to ask about origin key and not origin ID.

File: pkg/cmd/describe/origin/origin.go [hunk 4]
```diff
@@ -73,12 +73,12 @@ func NewCmd(f *cmdutil.Factory) *cobra.Command {
 					return utils.ErrorConvertingStringToInt
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:73]: You do not need to convert this. Origin key in not an integer!

File: pkg/cmd/describe/origin/origin.go [hunk 5]
```diff
@@ -107,7 +107,7 @@ func NewCmd(f *cmdutil.Factory) *cobra.Command {
 	}
 
 	cmd.Flags().Int64VarP(&applicationID, "application-id", "a", 0, msg.FlagApplicationID)
-	cmd.Flags().Int64VarP(&originID, "origin-id", "o", 0, msg.FlagOriginID)
+	cmd.Flags().Int64VarP(&originKey, "origin-key", "o", 0, msg.FlagOriginID)
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:110]: remove the short hand from both application-id and origin-key flags

**Review by PatrickMenoti — CHANGES_REQUESTED** [review]

**Comment by maxwelbm** [comment]
It's actually a string and I haven't refactored the origins part of the api to deal with the origin key, hot refactoring coming up kkk

**Review by LucasGeneroso — APPROVED** [review]

File: pkg/cmd/create/origin/origin.go [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+	}
+
+	if cmd.Flags().Changed("hmac-secret-key") {
+		request.SetHmacSecretKey(fields.HmacSecretKey)
+	}
+
+	return nil
+}
+
+func addFlags(flags *pflag.FlagSet, fields *Fields) {
```
Comment (PatrickMenoti) [inline pkg/cmd/create/origin/origin.go:179]: cool idea 🚀 
we just need to apply this to the other commands as well

File: pkg/cmd/describe/origin/origin.go [hunk 7]
```diff
@@ -1,51 +1,78 @@
...
 	"github.com/aziontech/azion-cli/pkg/cmdutil"
 	"github.com/aziontech/azion-cli/pkg/contracts"
+	"github.com/aziontech/azion-cli/pkg/logger"
 	"github.com/aziontech/azion-cli/utils"
 	sdk "github.com/aziontech/azionapi-go-sdk/edgeapplications"
 	"github.com/spf13/cobra"
 )
 
+var exemplo string = heredoc.Doc(`
+	$ azion origins describe --application-id 1673635839 --origin-key 0000000-00000000-00a0a00s0as0-000000
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:27]: change examples to `origin`

File: pkg/cmd/describe/origin/origin.go [hunk 8]
```diff
@@ -1,51 +1,78 @@
...
 	api "github.com/aziontech/azion-cli/pkg/api/origin"
 	"github.com/aziontech/azion-cli/pkg/cmdutil"
 	"github.com/aziontech/azion-cli/pkg/contracts"
+	"github.com/aziontech/azion-cli/pkg/logger"
 	"github.com/aziontech/azion-cli/utils"
 	sdk "github.com/aziontech/azionapi-go-sdk/edgeapplications"
 	"github.com/spf13/cobra"
 )
 
+var exemplo string = heredoc.Doc(`
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:26]: If you want to separate this, I think you could put this var inside the messages file. 
Also: change the var name to "example"

File: pkg/cmd/describe/origin/origin.go [hunk 9]
```diff
@@ -1,51 +1,78 @@
...
-      $ azion origins describe --application-id 1673635839 --origin-id 31223
-      $ azion origins describe --application-id 1673635839 --origin-id 31223--format json
-      $ azion origins describe --application-id 1673635839 --origin-id 31223--out "./tmp/test.json" --format json
-    `),
+		Example:       exemplo,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if !cmd.Flags().Changed("application-id") || !cmd.Flags().Changed("origin-id") {
-				return msg.ErrorMissingArguments
+			if !cmd.Flags().Changed("application-id") {
+				answers, err := utils.AskInput("What is the ID of the Edge Application?")
```
Comment (PatrickMenoti) [inline pkg/cmd/describe/origin/origin.go:48]: Put this input message inside a variable in the messages file
Also:
"What is the ID of the Edge Application this origin is linked to?"

**Review by PatrickMenoti — COMMENTED** [review]
Be careful with the order you merge your 2 PRs. 
This and the update PR both make changes to the Create Origin file.

File: pkg/cmd/describe/origin/origin.go (hunk 8 above)
Comment (maxwelbm) [reply pkg/cmd/describe/origin/origin.go:26]: I agree, let's put it in the messages to make it cleaner

**Review by PatrickMenoti — APPROVED** [review]

**Review by LucasGeneroso — APPROVED** [review]

**Pull request merged**